lib/ibanity/http_signature.rb

```diff
@@ -0,0 +1,87 @@
+module Ibanity
+  class HttpSignature
+    def initialize(certificate:, certificate_id:, key:, method:, uri:, query_params:, headers:, payload:)
+      @certificate    = certificate
+      @certificate_id = certificate_id
+      @key            = key
+      @method         = method
+      @uri            = URI(uri)
+      @headers        = headers
+      @payload        = payload
+      @query_params   = query_params
+    end
+
+    def payload_digest
+      digest         = OpenSSL::Digest::SHA512.new
+      string_payload = @payload.nil? ? "" : @payload.to_json
+      digest.update(string_payload)
+      base64 = Base64.urlsafe_encode64(digest.digest)
+      "SHA-512=#{base64}"
+    end
+
+    def signature_algorithm
+      @certificate.signature_algorithm.match("sha256") ? "rsa-sha256" : "rsa-sha512"
```

Comment:
Why do we have to support both sha256 and sha512?

Reply:
Because it depends on the signature of the certificate, has the certificate could be provided by a QTSP we have to support both.